File: backend/ResearchCruiseApp/Application/Models/DTOs/CruiseApplications/FormBDto.cs
using System.ComponentModel.DataAnnotations;

namespace ResearchCruiseApp.Application.Models.DTOs.CruiseApplications;


public class FormBDto
{
    [StringLength(1024)]
    public string IsCruiseManagerPresent { get; init; } = null!;

    public List<PermissionDto> Permissions { get; init; } = [];

    public List<UgTeamDto> UgTeams { get; init; } = [];

    public List<GuestTeamDto> GuestTeams { get; init; } = [];

    public List<CrewMemberDto> CrewMembers { get; init; } = [];

    public List<ShortResearchEquipmentDto> ShortResearchEquipments { get; init; } = [];

    public List<LongResearchEquipmentDto> LongResearchEquipments { get; init; } = [];

    public List<PortDto> Ports { get; init; } = [];

    public List<CruiseDayDetailsDto> CruiseDaysDetails { get; init; } = [];

    public List<ResearchEquipmentDto> ResearchEquipments { get; init; } = [];

    public List<Guid> ShipEquipmentsIds { get; init; } = [];
}

File: frontend/src/routes/applications/$applicationId/-schemas/types/FormBDto.ts
import { CrewMemberDto } from './CrewMemberDto';
import { CruiseDayDetailsDto } from './CruiseDayDetailsDto';
import { GuestTeamDto } from './GuestTeamDto';
import { LongResearchEquipmentDto } from './LongResearchEquipmentDto';
import { PermissionDto } from './PermissionDto';
import { PortDto } from './PortDto';
import { ResearchEquipmentDto } from './ResearchEquipmentDto';
import { ShortResearchEquipmentDto } from './ShortResearchEquipmentDto';
import { UGTeamDto } from './UGTeamDto';

export type FormBDto = {
  isCruiseManagerPresent: 'true' | 'false';
  permissions: PermissionDto[];
  ugTeams: UGTeamDto[];
  guestTeams: GuestTeamDto[];
  crewMembers: CrewMemberDto[];
  shortResearchEquipments: ShortResearchEquipmentDto[];
  longResearchEquipments: LongResearchEquipmentDto[];
  ports: PortDto[];
  cruiseDaysDetails: CruiseDayDetailsDto[];
  researchEquipments: ResearchEquipmentDto[];
  shipEquipmentsIds: string[];
};
